### PR TITLE
[DEV-2492]&[DEV-2494]&[DEV-2498]&[NOTASK] restrict swap, buyCrypto blockchainFee update, support issue fix, userData in notification

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations (on [DFX services](https://dev.services.dfx.swiss))
  - Login/logout
  - Buy/sell payment request
  - KYC page

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

File: src/subdomains/core/liquidity-management/services/liquidity-management-rule.service.ts [hunk 1]
```diff
@@ -51,18 +52,15 @@ export class LiquidityManagementRuleService {
...
     if (existingRule.status === LiquidityManagementRuleStatus.PROCESSING) {
       throw new BadRequestException('Rule is currently processing and cannot be updated');
     }
 
-    const rule = await this.checkAndCreateInstance(dto);
-    const updatedRule = await this.ruleRepo.save({ ...existingRule, ...rule });
-
-    return LiquidityManagementRuleOutputDtoMapper.entityToDto(updatedRule);
+    await this.ruleRepo.update(existingRule.id, { ...dto });
   }
```
Comment (davidleomay) [inline src/subdomains/core/liquidity-management/services/liquidity-management-rule.service.ts:64]:
```ts
await this.ruleRepo.update(existingRule.id, dto);
```

File: src/subdomains/core/trading/services/trading-rule.service.ts [hunk 2]
```diff
@@ -20,6 +21,13 @@ export class TradingRuleService {
 
   // --- PUBLIC API --- //
 
+  async updateTradingRule(id: number, dto: UpdateTradingRuleDto): Promise<void> {
+    const tradingRule = await this.ruleRepo.findOneBy({ id });
+    if (!tradingRule) throw new NotFoundException('Trading rule not found');
+
+    await this.ruleRepo.update(tradingRule.id, { ...dto });
+  }
```
Comment (davidleomay) [inline src/subdomains/core/trading/services/trading-rule.service.ts:29]: Same here

File: src/subdomains/core/history/mappers/transaction-dto.mapper.ts [hunk 3]
```diff
@@ -183,6 +187,33 @@ export class TransactionDtoMapper {
...
+        entity.totalFeeAmount != null &&
+        entity.blockchainFee != null &&
+        Util.roundReadable(
+          (entity.totalFeeAmount - entity.blockchainFee) * referencePrice,
+          isFiat(entity.inputAssetEntity),
+        ),
+      total:
+        entity.totalFeeAmount != null &&
+        Util.roundReadable(entity.totalFeeAmount * referencePrice, isFiat(entity.inputAssetEntity)),
+    };
```
Comment (davidleomay) [inline src/subdomains/core/history/mappers/transaction-dto.mapper.ts:215]: Haha, this is not working as expected 😉 
image

Comment (Yannick1712) [reply src/subdomains/core/history/mappers/transaction-dto.mapper.ts:215]: Hehe yeah already saw that 😅

**Review by davidleomay — APPROVED** [review]

**Pull request merged**